src/main/java/com/google/javascript/gents/ModuleRenameLogger.java

```diff
@@ -0,0 +1,51 @@
+package com.google.javascript.gents;
+
+import com.google.gson.Gson;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Generates a log that maps how goog.modules/goog.provides where mapped to TS modules.
+ *
+ * <p>The log contains: - the original goog.module/goog.provide name. - the name of the TypeScript
+ * file that contains the converted Closure module/namespace. - Gents never uses TS default exports.
+ * This means if the original goog.module had a default export we generated a named export with a
+ * specific name. The field 'defaultRename' contains that name. If there was no default export, the
+ * field contains the empty string.
+ *
+ * <p>Ex: (in file buz.js) goog.module(foo.bar) class A { ... } exports = A;
+ *
+ * <p>is translated to buz.ts: export class A { ... }
+ *
+ * <p>Which generates the following log line: foo.bar,buz.ts,A
+ */
+public class ModuleRenameLogger {
+  private class LogItem {
+    public String originalName;
+    public String tsFile;
+    public String defaultRename;
+
+    public LogItem(String originalName, String tsFile, String defaultRename) {
+      this.originalName = originalName;
+      this.tsFile = tsFile;
+      this.defaultRename = defaultRename;
+    }
+  }
+
+  private Gson gson = new Gson();
+
+  public String generateModuleRewriteLog(
+      Set<String> filesToConvert, Map<String, CollectModuleMetadata.FileModule> namespaceMap) {
+    StringBuilder builder = new StringBuilder();
+    for (String s : namespaceMap.keySet()) {
```

Comment:
prefer to use the entry set instead (peformance)

for (Entry<String, FileModule> entry : namespaceMap.entries())